### ISSUE TITLE
Deepsea grain does not get realized if it is set without value

### ISSUE THREAD

**Issue by Martin-Weiss** [SUSE/DeepSea#923]
When the deepsea grain is set without value - the targeting with 'salt "-G 'deepsea:*' test.ping"' works - but deepsea throws an error that no deepsea grain is set. If we set a value of "default" deepsea realizes the grain set.

It would be great if the mathing in deepsea and salt on the grain being set would be identical - so could we fix this by allowing the grain to exist also without value?

**Comment by swiftgist** [comment]
I believe this is a requirement of Salt, IIRC.  I was not able to set a key without a value.  Are you setting a value of ""?  I chose the value "default", although it's whole purpose is to guarantee that the grain is added and that it's somewhat obvious.

Part of me wants to leave the value for admin use completely.  I am undecided on that and have not had enough conversations.  The idea is giving admins yet another place to customize if needed.  If fairly large clusters of different hardware models, these could be set to specific values allowing the admin to deal with sub groups without changing the intended behavior of DeepSea.  (I think a more realistic example is setting the value to "trouble" for the few nodes in a cluster that need extra attention .)

**Comment by Martin-Weiss** [comment]
If I remember right I could set a grain without value by using 

`salt '*' grains.set deepsea`

and then the addressing with  

`salt "-G 'deepsea:' test.ping"`

works.

**Comment by swiftgist** [comment]
Okay, so you can set a grain without a value, but isn't that confusing?  Considering all grains have values and `salt '*' grains.get deepsea` returns the same as `salt '*' grains.get blueberry` (i.e. an absent key).  It's only apparent in a `grains.items` call and then it still looks odd.  I am not seeing the benefit of not typing one key.

**Comment by Martin-Weiss** [comment]
> Okay, so you can set a grain without a value, but isn't that confusing?

Yes - that might be confusing.. however - for targeting via the salt command this seems to be valid.
-G 'deepsea:*' works - and just the targeting in deepsea does not work.

So I am just thinking it would be great if we can adjust the deepsea targeting to work the same way as salt command targeting..
